pool.go

```diff
@@ -84,9 +84,7 @@ func (p *Pool) get(timeout time.Duration) *client {
 	default:
 	}
 
-	if p.created < p.max {
-		p.makeOne()
-	}
+	p.makeOne()
```

Comment:
Doesn't this result in unlimited amount of connections?

I believe that the `p.created < p.max` guard should ensure we only get the specified max connections.

Reply:
I believe (and correct me if I'm wrong) that `makeOne` should be safe to do.  It spins up a goroutine that:
* Calls inc()
  * inc() will check `p.created >= p.max` and if true, return false
* If inc() returns false nothing happens

So `makeOne` should not create a new connection unless `p.created < p.max`. The benefit to letting makeOne do the logic is that it handles the concurrency better -- it locks the mutex appropriately.

Reply:
Just checked, you are right.